### PR TITLE
API Implement Tika support

### PR DESCRIPTION
**Pull request by tractorcow**
API Implement support for detection via mime-type as well as file extension
API Implement FileContent property for safe usage in templates
API instead of returning the list of extensions / mime types supported, support is determined on a per-file bases
Marking dev-master as version 2.0 as this contains breaking changes

### CODE REVIEW

**Comment by tractorcow** [comment]
I've corrected the tests to work with 3.2 properly. :)

**Comment by tractorcow** [comment]
I've now updated the PR to correctly install both tika and pdfconvert for travis.

Turns out that the tika instaler was wrecking the PATH in the environment. took a while to fix. =/

File: code/extractors/PDFTextExtractor.php [hunk 1]
```diff
@@ -42,6 +74,24 @@ function getContent($path) {
...
\ No newline at end of file
+	/**
+	 * Removes utf-8 ligatures.
+	 *
+	 * @link http://en.wikipedia.org/wiki/Typographic_ligature#Computer_typesetting
+	 *
+	 * @param string $input
+	 * @return string
+	 */
+	protected function cleanupLigatures($input) {
```
Comment (chillu) [inline code/extractors/PDFTextExtractor.php]: Heh, good improvement. Did you just come across that edge case during testing?

Comment (tractorcow) [inline code/extractors/PDFTextExtractor.php]: While writing test cases. :) travis didn't have xpdf setup for testing so it was never covered.

**Comment by tractorcow** [comment]
Also see https://github.com/silverstripe-labs/silverstripe-fulltextsearch/pull/61 for using this with fulltextsearch

File: code/extractors/PDFTextExtractor.php (hunk 1 above)
Comment (tractorcow) [inline code/extractors/PDFTextExtractor.php]: I missed the `ﬅ` ligature. I'll add it now.

Comment (tractorcow) [inline code/extractors/PDFTextExtractor.php]: done:D

**Comment by chillu** [comment]
Fantastic job, thanks Damian! As discussed, should be followed up with a PR for allowing Tika in server mode.

**Comment by tractorcow** [comment]
Thanks @chillu , the server work is in progress. I'm copying the concept from another library I found at https://github.com/bangpound/tika-rest-client/tree/master/src/Bangpound/Tika, but maybe a bit stripped down.
